File: DynamicKey/AgoraDynamicKey/php/sample/EducationTokenBuilderSample.php
<?php
include("../src/EducationTokenBuilder.php");

// Need to set environment variable AGORA_APP_ID
$appId = getenv("AGORA_APP_ID");
// Need to set environment variable AGORA_APP_CERTIFICATE
$appCertificate = getenv("AGORA_APP_CERTIFICATE");

$expire = 600;
$roomUuid = "123";
$userUuid = "2882341273";
$role = 1;

echo "App Id: " . $appId . PHP_EOL;
echo "App Certificate: " . $appCertificate . PHP_EOL;
if ($appId == "" || $appCertificate == "") {
    echo "Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE" . PHP_EOL;
    exit;
}

$token = EducationTokenBuilder::buildRoomUserToken($appId, $appCertificate, $roomUuid, $userUuid, $role, $expire);
echo 'Education room user token: ' . $token . PHP_EOL;

$token = EducationTokenBuilder::buildUserToken($appId, $appCertificate, $userUuid, $expire);
echo 'Education user token: ' . $token . PHP_EOL;

$token = EducationTokenBuilder::buildAppToken($appId, $appCertificate, $expire);
echo 'Education app token: ' . $token . PHP_EOL;

File: DynamicKey/AgoraDynamicKey/ruby/sample/education_token_builder_sample.rb
require_relative '../lib/dynamic_key2'

# Need to set environment variable AGORA_APP_ID
app_id = ENV['AGORA_APP_ID']
# Need to set environment variable AGORA_APP_CERTIFICATE
app_certificate = ENV['AGORA_APP_CERTIFICATE']

room_uuid = '123'
user_uuid = '2882341273'
role = 1
token_expiration_in_seconds = 3600

puts "App Id: #{app_id}"
puts "App Certificate: #{app_certificate}"
if !app_id || app_id == '' || !app_certificate || app_certificate == ''
  puts 'Need to set environment variable AGORA_APP_ID and AGORA_APP_CERTIFICATE'
  exit
end

token = AgoraDynamicKey2::EducationTokenBuilder.build_room_user_token(app_id, app_certificate, room_uuid, user_uuid,
                                                                      role, token_expiration_in_seconds)
puts "Education room user token: #{token}"

token = AgoraDynamicKey2::EducationTokenBuilder.build_user_token(app_id, app_certificate, user_uuid,
                                                                 token_expiration_in_seconds)
puts "Education user token: #{token}"

token = AgoraDynamicKey2::EducationTokenBuilder.build_app_token(app_id, app_certificate, token_expiration_in_seconds)
puts "Education app token: #{token}"
